src/plugins/interactive_setup/public/manual_configuration_form.tsx

```diff
@@ -0,0 +1,436 @@
+  EuiIcon,
+  EuiLink,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import type { FunctionComponent } from 'react';
+import React from 'react';
+import useAsyncFn from 'react-use/lib/useAsyncFn';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import type { Certificate, PingResponse } from '../common/types';
+import type { ValidationErrors } from './use_form';
+import { useForm } from './use_form';
+import { useHttp } from './use_http';
+
+export interface ManualConfigurationFormValues {
+  host: string;
+  username: string;
+  password: string;
+  caCert: string;
+}
+
+export interface ManualConfigurationFormProps {
+  defaultValues?: ManualConfigurationFormValues;
+  onCancel(): void;
+  onSuccess?(): void;
+}
+
+export const ManualConfigurationForm: FunctionComponent<ManualConfigurationFormProps> = ({
+  defaultValues = {
+    host: 'https://localhost:9200',
+    username: 'kibana_system',
+    password: '',
+    caCert: '',
+  },
+  onCancel,
+  onSuccess,
+}) => {
+  const http = useHttp();
+
+  const [state, pingCluster] = useAsyncFn(async (values: ManualConfigurationFormValues) => {
+    const requiresCert = values.host.trim().startsWith('https://');
+    let requiresAuth = false;
+    let body: PingResponse;
+
+    try {
+      const response = await http.post<PingResponse>('/internal/interactive_setup/ping', {
+        body: JSON.stringify({
+          hosts: [values.host],
+        }),
+        asResponse: true,
+      });
+      body = response.body!;
+    } catch (error) {
+      if (error.response?.status !== 401) {
+        throw error;
+      }
+      requiresAuth = true;
+      body = error.body;
+    }
+
+    return { requiresCert, requiresAuth, ...body };
+  });
+
+  const [form, eventHandlers] = useForm({
+    defaultValues,
+    validate: async (values) => {
+      const errors: ValidationErrors<typeof values> = {};
+
+      let requiresAuth = false;
+      let requiresCert = false;
+      let needsToUploadCert = false;
+
+      if (!values.host) {
+        errors.host = i18n.translate('interactiveSetup.manualConfigurationForm.hostRequiredError', {
+          defaultMessage: 'Enter an address.',
+        });
+      } else {
+        try {
+          const url = new URL(values.host);
+          if (!url.protocol || !url.hostname || !url.port) {
+            throw new Error();
+          }
+          if (url.protocol === 'https:') {
+            requiresCert = true;
+          }
+        } catch (error) {
+          errors.host = i18n.translate(
+            'interactiveSetup.manualConfigurationForm.hostInvalidError',
+            {
+              defaultMessage: 'Enter a valid address including protocol and port number.',
+            }
+          );
+        }
+      }
+
+      if (!errors.host) {
+        try {
+          const data = await pingCluster(values);
+          if (data instanceof Error) {
+            throw data;
+          }
```

Comment:
__question:__ this check looks a bit weird, why do we need it?

Reply:
`useAsyncFn` hook returns the response error instead of throwing it so this will make sure we go to the catch statement in that case.